detection_rules/rule.py

```diff
@@ -1361,11 +1407,15 @@ def version_lock(self):
     def set_version_lock(self, value):
         from .version_lock import VersionLock
 
-        if value and not isinstance(value, VersionLock):
-            raise TypeError(f'version lock property must be set with VersionLock objects only. Got {type(value)}')
+        if RULES_CONFIG.bypass_version_lock:
+            print('Cannot set the version lock when the versioning strategy is configured to bypass the version '
```

Comment:
this should raise an assertion error just like the changes to `TOMLRuleContents`.set_version_lock` above